File: BackEnd/NodeEditor.RestAPI/NodeEditor.Entities/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NodeEditor.Entities
{
    public class Node:EntityBase
    {
        public NodeType NodeType { get; set; }
        public Guid NodeTypeId { get; set; }
        public ICollection<DataInputValue> DataInputValues { get; set; }

        public NodeGroup NodeGroup { get; set; }
        public Guid NodeGroupId { get; set; }

        public Guid? PreviousNodeId { get; set; }
        public Node? PreviousNode { get; set; }
        public ICollection<Node> NextNodes { get; set; }

    }
}

File: FrontEnd/node-editor/src/Models/Node.ts
import { NodeType } from "./NodeType";
import { DataInputValue } from "./DataInputValue";
import { NodeGroup } from "./NodeGroup";
import { EntityBase } from "./EntityBase";

export interface Node extends EntityBase {
    nodeType: NodeType;
    nodeTypeId: string;
    dataInputValues: DataInputValue[];
    nodeGroup: NodeGroup;
    nodeGroupId: string;
}